processor/configuration/src/main/java/com/redhat/service/smartevents/processor/resolvers/custom/GooglePubSubActionResolver.java

```diff
@@ -0,0 +1,31 @@
+package com.redhat.service.smartevents.processor.resolvers.custom;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import com.fasterxml.jackson.databind.node.TextNode;
+import com.redhat.service.smartevents.infra.models.gateways.Action;
+import com.redhat.service.smartevents.processor.GatewayConfiguratorService;
+import com.redhat.service.smartevents.processor.GatewayResolver;
+import com.redhat.service.smartevents.processor.actions.google.GooglePubSubAction;
+import com.redhat.service.smartevents.processor.actions.kafkatopic.KafkaTopicAction;
+
+@ApplicationScoped
+public class GooglePubSubActionResolver implements GooglePubSubAction, GatewayResolver<Action> {
+
+    @Inject
+    GatewayConfiguratorService gatewayConfiguratorService;
+
+    @Override
+    public Action resolve(Action action, String customerId, String bridgeId, String processorId) {
+
+        Action resolvedAction = new Action();
+        resolvedAction.setParameters(action.getParameters().deepCopy());
+        resolvedAction.setType(KafkaTopicAction.TYPE);
+
+        String connectorTopicName = gatewayConfiguratorService.getConnectorTopicName(processorId);
+        resolvedAction.getParameters().set(KafkaTopicAction.TOPIC_PARAM, new TextNode(connectorTopicName));
```

Comment:
I think you need to use the `InternalKafkaConnectionPayload` class like [here](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/blob/main/processor/configuration/src/main/java/com/redhat/service/smartevents/processor/resolvers/custom/AwsLambdaActionResolver.java#L26).

Btw we probably need to refactor this with a default impl because it looks to me exactly a copy of the other Connector based actions.
Wdyt?
Feel free to create a follow up ticket

Reply:
See https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/888

Reply:
done